### PR TITLE
nucleotide-count: improve error handling fixes #298

### LINKED ISSUE #298: nucleotide-count: Improve error-handling behavior
As part of #275, we'd like to more closely follow idiomatic Go behavior in the face of errors.

If the nucleotide-count exercise does not have error handling, then this ticket can be closed immediately.

If it does, then follow the below steps:
#### Ensure `err` return value is of type `error`

It's idiomatic for the error return value to be of type `error`.

Most tests simply check whether `err == nil` or not. If we don't verify the type of `err`, it could be any type that has nils.

To lean on the type system, add the following line after assigning err:

``` go
var _ error = err
```

See [the hamming tests](https://github.com/exercism/xgo/blob/master/exercises/hamming/hamming_test.go#L14) for an example.
#### Ignore the normal return value when an error expected

The contract in Go is "if there's an error, assume nothing about the normal return value". Thus, we should only check the non-error return value when there is no error.

In some exercises we found that we were checking the normal return value, even when there was an error.

See [the binary tests](https://github.com/exercism/xgo/blob/master/exercises/binary/binary_test.go#L41-L58) for an example of the behavior that we want.
#### Increment the test version

These changes break backwards compatibility. Bump the test version to let reviewers know that the exercise has changed.

This requires changing two constants:
- `targetTestVersion` in the test file (`*_test.go`)
- `testVersion` in the reference solution (`example.go`)

If the exercise does not have a test version, then add one, setting the value to 1.

See [the version test in leap](https://github.com/exercism/xgo/blob/master/exercises/leap/leap_test.go#L13-L15) for an example.

### PR DESCRIPTION
**Pull request by ferhatelmas**
In addition to improving error handling, I made some changes according to my taste such as map equality check, value printing, etc.

Unless well aligned, I can revert or update w.r.t your comments. Thanks!

### CODE REVIEW

**Comment by ferhatelmas** [comment]
Ah, just saw that there is already an open PR :disappointed:

**Comment by kytrinyx** [comment]
It doesn't look like the author of the other open PR has come back, so I'm happy to push this one forward.

There are a number of changes in here that are not about improving the error handling. I do like the changes—would you be willing to open a separate PR for those and only clean up the error handling in this one?

**Comment by kytrinyx** [comment]
(If you open a PR for that, would you `@`-mention me? I've got a lot going on and could definitely use the specific reminder).

**Comment by ferhatelmas** [comment]
@kytrinyx I removed irrelevant parts from this PR. After this get merged, I will create another PR for other improvements. Thanks!

File: exercises/nucleotide-count/nucleotide_count_test.go [hunk 1]
```diff
@@ -79,6 +83,10 @@ var histogramTests = []histogramTest{
 		Histogram{'A': 0, 'C': 0, 'T': 0, 'G': 8},
 	},
 	{
+		"GGXXGGGG", // skip mutation
```
Comment (petertseng) [inline exercises/nucleotide-count/nucleotide_count_test.go]: A new case I am a little unsure about! Made harder by the fact that `DNA` doesn't return an `error`. Should it? That (plus this change) might be for a separate PR though.

This is difficult for me to imagine, since x-common doesn' thave canonical-data.json for nucleotide-count. Have other tracks with nucleotide-count similar cases to this?

File: exercises/nucleotide-count/nucleotide_count_test.go [hunk 2]
```diff
@@ -34,20 +34,17 @@ var tallyTests = []struct {
 func TestNucleotideCounts(t *testing.T) {
 	for _, tt := range tallyTests {
 		dna := DNA(tt.strand)
-		count, _ := dna.Count(tt.nucleotide)
-		if count != tt.expected {
+		if count, err := dna.Count(tt.nucleotide); err != nil {
+			t.Fatal(err)
+		} else if count != tt.expected {
```
Comment (cpuguy83) [inline exercises/nucleotide-count/nucleotide_count_test.go]: Can drop the else since the above is a `Fatal` and will return immediately.

File: exercises/nucleotide-count/nucleotide_count_test.go (hunk 1 above)
Comment (ferhatelmas) [inline exercises/nucleotide-count/nucleotide_count_test.go]: Removed this test case. 

I was planning to open another after this one for cleaning as suggested by @kytrinyx [here](https://github.com/exercism/xgo/pull/407#issuecomment-255615703). I can include error and test case there.

**Comment by ferhatelmas** [comment]
@petertseng Removed test case and rebased on master.

File: exercises/nucleotide-count/nucleotide_count_test.go (hunk 2 above)
Comment (petertseng) [inline exercises/nucleotide-count/nucleotide_count_test.go]: will have to watch though, `count` might not be in scope since now `count, err :=` is in the `if`.

**Comment by petertseng** [comment]
All right! Thanks for the work.
